common/src/main/java/tds/common/entity/utils/ChangeListener.java

```diff
@@ -0,0 +1,8 @@
+package tds.common.entity.utils;
+
+/**
+ * An interface for listening for and responding to changes on an entity
+ */
+public interface ChangeListener<T> {
+    void accept(T oldInstance, T newInstance);
```

Comment:
Hrm... Is Spring able to inject ChangeListeners based upon the T generic?  I can't remember if I've ever tried injecting-by-generic before...